### DISCUSSION TITLE
Toggleable Filters in Upgrades

### DISCUSSION THREAD

**Discussion by desagas** [P3pp3rF1y/SophisticatedBackpacks#153] [Ideas]
Just like each upgrade is toggleable in the backpack, I would love it if we could also toggle the filter for the upgrades.

Why? There are many times where I want my magnet to pick up everything around me, but, there are also times where I want it to not pickup or pick up something specific, and, in order to go back and forth, I have to either have two magnet upgrades, or, I have to repopulate each filter every time I want to toggle this.

This could also work for compacting, magnet, and many other upgrades.

Note, I would not want to toggle each item in a filter, only the entire filter itself.

**Comment by digitalbarrito** [comment]
+1 to this idea

Figured I'd chime in on this since it's Filter Upgrade related as well, but stackable filter upgrades.

Meaning you could slot say, 2 Adv. Filter Upgrades in the same bag, set them both to block the output of specific items, and have their filters work in conjunction making one large filter block the output of both filter lists. Alternatively, increase the maximum filter size to 40 or 60 in the config but leave the default configuration as is?

Why? Well, I use backpacks as better chests in an Auto-Sorting Storage system connected with Pipes, and it'd be wonderful to have a backpack that's able to specify more than 20 items max in a filter, and with the current way filters work, if you add a second filter upgrade, the filters on the first upgrade are ignored entirely.

**Reply by P3pp3rF1y** [reply]
technically they are not entirely ignored - if you specify to only allow specific items you can use multiple filters together to specify all of the items that are allowed. This is because it evaluates all filters so as soon as it finds the first filter which allows something to go in the backpack it will allow it to. Doesn't work with blocking though because as you say it evaluates the filters individually. 
However I don't see a good way of making this to work - the issue is that multiple filters can have different settings as to whether they block/allow, filter by nbt, tags, damage. And because of that these definitely cannot be evaluated together if they are both allow and block and there are probably very awkward things going to happen if the other settings are different between filters.
So likely the only solution would be to only have filters work together if all of their settings are identical and they differ only in what stacks they have in ther filter slots, but then I find it very difficult to communicate to the player that filters are sometimes working together and sometimes not. There would definitely need to be an obvious way of showing that is or is not the case otherwise it would just be a source of confusion and reports that things don't work the way people expect them to.
